Spatial.gd

```diff
@@ -692,25 +692,32 @@ func gen_adjustment_nodes():
 				continue
 			var new_gizmo = gizmo_scene.instance()
 			new_gizmo.translation = gizmo_position
-			new_gizmo.link_point("path", route, path2d, i)
-			new_gizmo.connect("selected", self, "on_gizmo_selected")
-			new_gizmo.connect("deselected", self, "on_gizmo_deselected")
+			new_gizmo.connect("selected", self, "on_gizmo_selected", [route, "path", path2d, i])
+			new_gizmo.connect("deselected", self, "on_gizmo_deselected", [route, "path", path2d, i])
+			new_gizmo.connect("updated", route, "update_point_poistion", [i])
+			new_gizmo.connect("updated", path2d, "update_point_poistion", [i])
 			$Gizmos.add_child(new_gizmo)
 	for icon in category3d.icons:
 		var new_gizmo = gizmo_scene.instance()
 		new_gizmo.translation = icon.translation
-		new_gizmo.link_point("icon", icon)
-		new_gizmo.connect("selected", self, "on_gizmo_selected")
-		new_gizmo.connect("deselected", self, "on_gizmo_deselected")
+		new_gizmo.connect("selected", self, "on_gizmo_selected", [icon, "icon"])
+		new_gizmo.connect("deselected", self, "on_gizmo_deselected", [icon, "icon"])
+		new_gizmo.connect("updated", icon, "update_point_poistion")
 		$Gizmos.add_child(new_gizmo)
 
+var currently_selected_gizmo = null
+var currently_selected_node3d = null
+var currently_selected_node2d = null
+var currently_selected_index = 0
 
-var currently_selected_node = null
-func on_gizmo_selected(object):
-	self.currently_selected_node = object
+func on_gizmo_selected(object, node3d, point_type: String, node2d = null, index: int = 0):
```

Comment:
By splitting this function into unique functions for path and icon, I can eliminate `point_type` and the use of optional variables. This will also make the typing possible.